examples/linear_model/plot_lasso_model_selection.py

```diff
@@ -1,171 +1,232 @@
+=================================================
 
+This example focuses on model selection for Lasso models that are
+linear models with a L1 penalty for regression problems.
+
+Indeed, several strategies can be used to select the value of the
+regularization parameter: using an information criterion, namely AIC or BIC or
+via cross-validation.
+
+In what follows, we will discuss in details the different strategies.
 """
 
-# Author: Olivier Grisel, Gael Varoquaux, Alexandre Gramfort
+# Author: Olivier Grisel
+#         Gael Varoquaux
+#         Alexandre Gramfort
+#         Guillaume Lemaitre
 # License: BSD 3 clause
 
-import time
+# %%
+import sklearn
 
-import numpy as np
-import matplotlib.pyplot as plt
+sklearn.set_config(display="diagram")
 
-from sklearn.linear_model import LassoCV, LassoLarsCV, LassoLarsIC
-from sklearn import datasets
+# %%
+# Dataset
+# -------
+# In this example, we will use the diabetes dataset.
+from sklearn.datasets import load_diabetes
 
-# This is to avoid division by zero while doing np.log10
-EPSILON = 1e-4
+X, y = load_diabetes(return_X_y=True, as_frame=True)
+X.head()
 
-X, y = datasets.load_diabetes(return_X_y=True)
+# %%
+# In addition, we will add some random features to the original data to
+# make obvious the feature selection performed by the Lasso model.
+import numpy as np
+import pandas as pd
 
 rng = np.random.RandomState(42)
-X = np.c_[X, rng.randn(X.shape[0], 14)]  # add some bad features
-
-# normalize data as done by Lars to allow for comparison
-X /= np.sqrt(np.sum(X ** 2, axis=0))
-
-# #############################################################################
-# LassoLarsIC: least angle regression with BIC/AIC criterion
-
-model_bic = LassoLarsIC(criterion="bic", normalize=False)
-t1 = time.time()
-model_bic.fit(X, y)
-t_bic = time.time() - t1
-alpha_bic_ = model_bic.alpha_
-
-model_aic = LassoLarsIC(criterion="aic", normalize=False)
-model_aic.fit(X, y)
-alpha_aic_ = model_aic.alpha_
-
-
-def plot_ic_criterion(model, name, color):
-    criterion_ = model.criterion_
-    plt.semilogx(
-        model.alphas_ + EPSILON,
-        criterion_,
-        "--",
-        color=color,
-        linewidth=3,
-        label="%s criterion" % name,
-    )
-    plt.axvline(
-        model.alpha_ + EPSILON,
-        color=color,
-        linewidth=3,
-        label="alpha: %s estimate" % name,
-    )
-    plt.xlabel(r"$\alpha$")
-    plt.ylabel("criterion")
-
-
-plt.figure()
-plot_ic_criterion(model_aic, "AIC", "b")
-plot_ic_criterion(model_bic, "BIC", "r")
-plt.legend()
-plt.title("Information-criterion for model selection (training time %.3fs)" % t_bic)
-
-# #############################################################################
-# LassoCV: coordinate descent
+n_random_features = 14
+X_random = pd.DataFrame(
+    rng.randn(X.shape[0], n_random_features),
+    columns=[f"random_{i:02d}" for i in range(n_random_features)],
+)
+X = pd.concat([X, X_random], axis=1)
+X.head()
```

Comment:
This table has too many columns to properly render. Maybe cut off after the first random feature?